internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -270,18 +271,19 @@ func (r *Reconciler) getMachineSetsForDeployment(ctx context.Context, d *cluster
 
 		// If a MachineDeployment with a nil or empty selector creeps in, it should match nothing, not everything.
 		if selector.Empty() {
-			log.Info("Skipping MachineSet as the selector is empty")
+			log.V(5).Info("Skipping MachineSet as the selector is empty")
 			continue
 		}
 
 		// Skip this MachineSet unless either selector matches or it has a controller ref pointing to this MachineDeployment
 		if !selector.Matches(labels.Set(ms.Labels)) && !metav1.IsControlledBy(ms, d) {
-			log.V(4).Info("Skipping MachineSet, label mismatch")
+			log.V(5).Info("Skipping MachineSet, label mismatch")
 			continue
 		}
 
 		// Attempt to adopt machine if it meets previous conditions and it has no controller references.
 		if metav1.GetControllerOf(ms) == nil {
+			log.Info(fmt.Sprintf("Adopting MachineSet %q into MachineDeployment %q", ms.Name, d.Name))
```

Comment:
[MachineDeployment should be part of the logger already](https://github.com/kubernetes-sigs/controller-runtime/blob/1e4d87c9f9e15e4a58bb81909dd787f30ede7693/alias.go#L134-L141) so `Adopting MachineSet %q` should be enough here.